### PR TITLE
Add new attributes for memory and dimm devices.

### PR DESCRIPTION
**Pull request by YongxueHong**
Add new attributes for memory and dimm devices.

ID: 2054732

### CODE REVIEW

**Comment by YongxueHong** [comment]
Hi, @zhenyzha @MiriamDeng @yanan-fu @fbq815 @huangyum @luckyh 
Can you help me to review it, thanks.

**Comment by huangyum** [comment]
@mcasquer Would you please help review, thanks.

**Comment by mcasquer** [comment]
As far as I can see, here is being reverted changes from: avocado-framework/avocado-vt#3285

Based on that changes I created this other PR that now is merged: autotest/tp-qemu#3188
@YongxueHong should this be reverted too?

**Comment by YongxueHong** [comment]
> As far as I can see, here is being reverted changes from: #3285
> 
> Based on that changes I created this other PR that now is merged: [autotest/tp-qemu#3188](https://github.com/autotest/tp-qemu/pull/3188) @YongxueHong should this be reverted too?

Hi, @mcasquer 
Yes, we should reverted it too, you can revert it after this PR is merged.
Sorry for inconvenience.

**Review by fbq815 — APPROVED** [review]
On s390x:
RHEL8:
RESULTS    : PASS 64 | ERROR 2 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2022-03-01T06.26-3d8f7e5/results.html
JOB TIME   : 10276.45 s

RHEL9:
RESULTS    : PASS 64 | ERROR 2 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2022-03-01T06.26-3d8f7e5/results.html
JOB TIME   : 10276.45 s

LGTM

**Comment by mcasquer** [comment]
> Hi, @mcasquer
> Yes, we should reverted it too, you can revert it after this PR is merged.
> Sorry for inconvenience.

Not a problem :grinning: 
I will update tests results when they finish

**Comment by zhenyzha** [comment]
On aarch64:
RHEL9
`numa.numa_stress,virtio_fs_share_data.default.default.with_cache.auto.default.default,hotplug_memory,vsock_test.seqpacket_test`

All Pass

Update RHEL8 test results later

**Comment by MiriamDeng** [comment]
The test is going on ppc platform, once it finishes test results will be updated to the patch, thanks a lot.

**Comment by mcasquer** [comment]
On x86_64:

RHEL9 & RHEL8.6
All hotplug_memory tests passed

LGTM

**Comment by YongxueHong** [comment]
Hi, @huangyum and @yanan-fu 
Do you have any comments?

**Review by yanan-fu — APPROVED** [review]
The logic is ok for me, but i did not check the new attributes details,  I think feaure owner have verify it, so ACK.

**Comment by zhenyzha** [comment]
On aarch64:
RHEL8
`numa.numa_stress,virtio_fs_share_data.default.default.with_cache.auto.default.default,hotplug_memory,vsock_test.seqpacket_test`

All Pass

LGTM.Acked-by: Zhenyu Zhang <[zhenyzha@redhat.com](mailto:zhenyzha@redhat.com)>

**Review by zhenyzha — APPROVED** [review]

File: virttest/qemu_devices/qdevices.py [hunk 1]
```diff
@@ -1487,13 +1504,13 @@ def verify_hotplug(self, out, monitor):
 class Dimm(QDevice):
     """
     PC-Dimm or NVDIMM device, support for memory hotplug using the device and
-    QOM objects.
+    the QOM objects 'memory-backend-ram' and 'memory-backend-file'
```
Comment (huangyum) [inline virttest/qemu_devices/qdevices.py:1507]: And 'memory-backend-memfd'.

**Review by huangyum — COMMENTED** [review]
Others look good to me.

File: virttest/qemu_devices/qdevices.py (hunk 1 above)
Comment (YongxueHong) [reply virttest/qemu_devices/qdevices.py]: Updated, thanks.

**Comment by YongxueHong** [comment]
Thanks all, let's merge it.

**Pull request merged**